packages/next-intl/test/middleware/middleware.test.tsx

```diff
@@ -68,6 +69,73 @@ beforeEach(() => {
   vi.clearAllMocks();
 });
 
+it.only('has docs that suggest a reasonable default matcher', () => {
+  // Approach 1: Allow-list certain pathnames to contain dots
+  const matcherFromDocs = [
+    // Skip certain segments and all files with an extension (e.g. favicon.ico)
+    '/((?!_next|api|_vercel|.+\\..+).*)',
+    // Allow all characters, including dots (e.g. /users/jane.doe)
+    '/(.+)?/users/(.+)'
+  ];
+
+  // Approach 2: Be restrictive with static assets
+  // Note: Seems too difficult, also due to restrictions from path-to-regexp)
+  // const matcherFromDocs = [
+  //   '/((?!api|_next|_vercel|favicon.ico|.*\\.txt|.*\\.png|.*\\.ico|.*\\.jpg|.*\\.jpeg|.*\\.svg|.*\\.json|.*\\.webmanifest|.*\\.gif|.*\\.xml|.*\\.webp|.*\\.html).*)'
+  // ];
+
+  // Approach 3: Ask user to put all static assets in something like `/public/assets`
+  // Note: Doesn't work well with built-in metadata files support from Next.js
+
+  // Approach 4: Handle the matching within the middleware, skip processing of known extensions,
+  // but allow the user to disable this (e.g. `createMiddleware({matcher: false})`)
+  // See https://github.com/shuding/nextra/blob/feed42a2dbdcd0d15d6c64bd3beab400044a6977/packages/nextra/src/locales.ts#L41
```

Comment:
After some thinking I came to somewhat the same conclusion that if you want to exclude URLs that contain file extensions you need to define explicitly what it needs to look for like the regex you linked, the problem with this tho is that you can never have all available file extensions.

Like how does this piece of code handle webp images? It doesn't. So you need to keep updating this list.
https://github.com/shuding/nextra/blob/feed42a2dbdcd0d15d6c64bd3beab400044a6977/packages/nextra/src/locales.ts#L41

I was thinking maybe it is better to use the headers from the request instead of the URL, maybe it's possible to check for content-type or something.

Reply:
> you can never have all available file extensions

That's correct, e.g. based on the tests I currently have in this file, it's easy to run into a false negative.

We could aggregate some lists from https://github.com/sindresorhus/is-image, https://github.com/sindresorhus/binary-extensions, https://github.com/sindresorhus/text-extensions etc., but looking through the lists, it would likely result in a situation where it's too unpredictable if the user will run into a false positive.

After all, if you have a path like `/users/jane.z`, you don't want to worry if `z` is a valid file extension. Even `.jpg` seems too risky there.



> I was thinking maybe it is better to use the headers from the request instead of the URL, maybe it's possible to check for content-type or something.

I also had that in mind some time ago, but I'm not sure it can be done in a reliable way.
 E.g. for an embedded image in a web page, in Chrome I see `Accept:
image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8` as the request header. However when you request the image directly in the browser, you'll get `Accept:
text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7`. Would be really odd if you right click + open an image and then it isn't displayed anymore.

On a second thought, maybe approach number 1 is more suitable and we should educate users about this early, before they run into an issue. The biggest tradeoff there IMO is that users could forget to update the matcher when a route is added that accepts a param that can potentially include a dot.

It seems like this is a situation where there's always a tradeoff involved and we have to pick the one with the least chance that users run into an issue.

Reply:
> It seems like this is a situation where there's always a tradeoff involved and we have to pick the one with the least chance that users run into an issue.

Yeah, I guess you're right. There is no solution that doesn't have a tradeoff, and approach 1 seems to be a good one. I had my hopes up for using the Accept header, but not being able to open it in a new tab is a deal breaker.

I am currently using approach 2 as it works for my usecase at this moment. 

We are not alone
https://github.com/vercel/next.js/discussions/36308#discussioncomment-6776963

Reply:
That seems reasonable to me. I've added a section in the docs in regard to the middleware matcher and linked to it from the getting started docs.

Would you like to have a look and let me know if you think it's helpful? [Proposed docs](https://next-intl-docs-git-fix-504-middleware-matcher-regex-next-intl.vercel.app/)

Many thanks for investigating and reporting this issue again! Hopefully this can save future users some headaches.